### ISSUE TITLE
GC Warning: Bad initial heap size 100000 - ignoring it.

### ISSUE THREAD

**Issue by davidak** [NixOS/nixpkgs#43339]
## Issue description

I got the warning multiple times when installing NixOS from `nixos-minimal-18.09pre145524.2a8a5533d18-x86_64-linux.iso`.

### Steps to reproduce

```
[root@nixos:~]# nix-channel --update
unpacking channels...
GC Warning: Bad initial heap size 100000 - ignoring it.
created 2 symlinks in user environment

[root@nixos:~]# nixos-install
building the configuration in /mnt/etc/nixos/configuration.nix...
GC Warning: Bad initial heap size 100000 - ignoring it.
[166 built, 377 copied (1244.9 MiB), 8.7 MiB DL]
GC Warning: Bad initial heap size 100000 - ignoring it.
copying channel...
GC Warning: Bad initial heap size 100000 - ignoring it.
installing the boot loader...
setting up /etc...
Initializing machine ID from random generator.
updating GRUB 2 menu...
installing the GRUB 2 boot loader on /dev/sda...
Installing for i386-pc platform.
Installation finished. No error reported.
setting up /etc...
setting root password...
New password: 
Retype new password: 
passwd: password updated successfully
installation finished!
```

## Technical details

[root@nixos:~]# nix-shell -p nix-info --run "nix-info -m"
GC Warning: Bad initial heap size 100000 - ignoring it.
GC Warning: Bad initial heap size 100000 - ignoring it.
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.53, NixOS, 18.09pre145524.2a8a5533d18 (Jellyfish)`
 - multi-user?: `no`
 - sandbox: `yes`
GC Warning: Bad initial heap size 100000 - ignoring it.
 - version: `nix-env (Nix) 2.0.4`
GC Warning: Bad initial heap size 100000 - ignoring it.
 - channels(root): `"nixos-18.09pre145524.2a8a5533d18"`
GC Warning: Bad initial heap size 100000 - ignoring it.
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by charles-dyfis-net** [comment]
Out of curiosity -- what's the hardware you've seen this on? Unless it's a recent regression that happens even on smaller boxes, I first saw this when installing 18.09 on a machine with a somewhat ridiculously large amount of RAM.

**Comment by pukkamustard** [comment]
I get the same warning running on a Rasberry Pi 3 (aarch64).

Warning is coming from `libgc` (https://github.com/ivmai/bdwgc/blob/c50de12ab045b57953e152545836c23f76e7bc24/misc.c#L1289) and seems to be caused by this option: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/profiles/installation-device.nix#L66

**Comment by Xel** [comment]
I got the same warning when installing 18.09 on a Thinkpad T570 with 16GB ram. and as @pukkamustard correctly stated seems to be caused by https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/profiles/installation-device.nix#L66

**Comment by c0bw3b** [comment]
Seen this just now on a test VM with no storage, just running a 18.09-minimal live CD.
The VM has 1 vCPU and 1GB vRAM.

**Comment by c0bw3b** [comment]
Solved in #49328
